datafusion/sql/src/unparser/plan.rs

```diff
@@ -733,10 +733,10 @@ impl Unparser<'_> {
                 // So we will append the alias to this subquery.
                 // Example:
                 //   select * from t1 limit 10 -> (select * from t1 limit 10) as a
-                if alias.is_some()
-                    && (table_scan.projection.is_none() && table_scan.filters.is_empty())
-                {
-                    builder = builder.alias(alias.clone().unwrap())?;
+                if let Some(alias) = alias {
+                    if table_scan.projection.is_none() && table_scan.filters.is_empty() {
+                        builder = builder.alias(alias.clone())?;
```

Comment:
```suggestion
                        builder = builder.alias(alias)?;
```

I don't believe this final clone is needed, since its the last place to use it.